onnxruntime/core/framework/session_state_utils.cc

```diff
@@ -161,9 +161,11 @@ static common::Status DeserializeTensorProto(const Env& env, const std::basic_st
     }
 
     OrtCallback ext_data_deleter;
+    ScopedOrtCallbackInvoker scoped_ort_callback_invoker;
     if (utils::HasExternalData(tensor_proto)) {
```

Comment:
Could you use std::optional to avoid the changes to callback.h?

Reply:
I can, but in general RAII classes should also have a default constructor (like unique_ptr, shared_ptr, scoped_ptr) and a way to either reset its contents or copy/move them. Hence the change to callback.h.